src/main/java/de/tum/in/www1/artemis/domain/push_notification/PushNotificationDeviceConfiguration.java

```diff
@@ -0,0 +1,94 @@
+package de.tum.in.www1.artemis.domain.push_notification;
+
+import java.util.Date;
+
+import javax.persistence.*;
+
+import org.hibernate.annotations.Cache;
+import org.hibernate.annotations.CacheConcurrencyStrategy;
+import org.hibernate.annotations.OnDelete;
+import org.hibernate.annotations.OnDeleteAction;
+
+import de.tum.in.www1.artemis.domain.User;
+
+/**
+ * Model for saving PushNotification DeviceTokens for native clients
+ */
+@Entity
+@Table(name = "push_notification_device_configuration")
+@Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
+@IdClass(PushNotificationDeviceConfigurationId.class)
+public class PushNotificationDeviceConfiguration {
+
+    @Id
+    @Column(name = "token")
+    private String token;
+
+    @Id
+    @Column(name = "device_type")
+    private PushNotificationDeviceType deviceType;
+
+    @Column(name = "expiration_date")
+    private Date expirationDate;
+
+    @Column(name = "secret_key")
+    private byte[] secretKey;
+
+    @Id
+    @ManyToOne(cascade = CascadeType.ALL)
```

Comment:
I don't think you want to cascade here, the user should not be deleted, if the PushNotificationDeviceConfiguration is deleted. In fact, you need to make sure that the PushNotificationDeviceConfiguration is deleted, when the user is deleted.

I would consider making this relationship bidirectional